pkg/common/cluster/clusterutil.go

```diff
@@ -0,0 +1,145 @@
+package cluster
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/hashicorp/go-multierror"
+	osconfig "github.com/openshift/client-go/config/clientset/versioned"
+	"github.com/openshift/osde2e/pkg/common/cluster/healthchecks"
+	"github.com/openshift/osde2e/pkg/common/config"
+	"github.com/openshift/osde2e/pkg/common/metadata"
+	"github.com/openshift/osde2e/pkg/common/spi"
+	"github.com/openshift/osde2e/pkg/common/state"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+const (
+	// errorWindow is the number of checks made to determine if a cluster has truly failed.
+	errorWindow = 5
+)
+
+// WaitForClusterReady blocks until the cluster is ready for testing.
+func WaitForClusterReady(provisioner spi.Provisioner, clusterID string) error {
```

Comment:
Should waiting for cluster readiness be part of the provisioner?

Reply:
IMO no, and the reason is that we need direct access to the cluster in order to assess whether a cluster is ready -- because there's a difference between OCM's notion of "ready" and our notion of "ready." I don't think the provisioner should be doing direct cluster access here, just letting us know what the provisioner APIs are reporting back.

If OCM did osd-verify, I'd be fine with it, but since that doesn't seem to be the case, I'd rather extract it out. Thoughts?